File: usage-case/github/src/main/java/org/luvx/github/vavr/TupleCase.java
package org.luvx.github.vavr;

import io.vavr.Tuple;

public class TupleCase {
    public static void main(String[] args) {
        Tuple.of("Hello", 100)
                .map(String::toLowerCase, v -> v * 5)
                .apply((s, n) -> s + n)
        ;
    }
}

File: usage-case/api/src/main/kotlin/org/luvx/coding/vavr/TupleCase.kt
package org.luvx.coding.vavr

import io.vavr.Tuple
import org.luvx.coding.common.enhancer.out

fun main() {
    Tuple.of("Hello", 100)
        .map({ obj -> obj.lowercase() }) { v -> v * 5 }
        .apply { s, n -> s + n }
        .out()
}